src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/SharedTableBuilder.java

```diff
@@ -318,24 +322,28 @@ public MtAmazonDynamoDbBySharedTable build() {
         setDefaults();
         withName("SharedTableBuilder");
         validate();
+        PhysicalTableManager physicalTableManager = new PhysicalTableManager(amazonDynamoDb, pollIntervalSeconds);
         if (tableMappingFactory == null) {
             CreateTableRequestFactory createTableRequestFactory = new SharedTableCreateTableRequestFactory(
-                partitioningStrategy.getTablePrimaryKeyMapper(), createTableRequests, getTablePrefix());
+                partitioningStrategy.getTablePrimaryKeyMapper(), createTableRequests,
+                partitioningStrategy::toCompatiblePhysicalPrimaryKey, tablePrefix,
```

Comment:
TablePartitioningStrategy now has a new method toCompatiblePhysicalPrimaryKey() that converts a given virtual PK  into a physical PK, so that when there's a request to create a new multitenant table, we know how to convert the virtual table CreateTableRequest into a physical one.